### PR TITLE
integrate_memberlist_column hook

### PR DESCRIPTION
**Pull request by Garthi**
for easy manipulation the member list with a Mod

see the example: https://gist.github.com/Garthi/9cb704cbd39e6a884611

### CODE REVIEW

**Comment by Oldiesmann** [comment]
@Garthi Please sign-off on this commit so we can use it.

File: Sources/Memberlist.php [hunk 1]
```diff
@@ -212,6 +224,8 @@ function MLAll()
 	// 3. the page shown is high enough to make a DB filesort unprofitable.
 	$use_cache = $modSettings['totalMembers'] > 2000 && (!isset($_REQUEST['sort']) || $_REQUEST['sort'] === 'real_name') && isset($_REQUEST['start']) && $_REQUEST['start'] > $cache_step_size;
 
+	$memberlist_cache = Array();
```
Comment (norv) [inline Sources/Memberlist.php:227]: This doesn't seem necessary, the only use of $memberlist_cache is in the next conditional and it is initialized in all cases, although I agree that it's not obvious. (and not very cool either)

On another note, SMF uses array(), as a matter of code style.

File: Themes/default/Memberlist.template.php [hunk 2]
```diff
@@ -67,63 +67,78 @@ function template_main()
...
+					<td class="', $class, '">', $member[$key]['link'], '</td>';
+						
+					}
+				}
+			}
+			
+			echo '
 				</tr>';
-
+
```
Comment (norv) [inline Themes/default/Memberlist.template.php:141]: This and other lines have useless tabs.

**Comment by emanuele45** [comment]
Just a quick note, the hook is not **that** useful:
- the array passed is already in the global scope,
- few lines below there is already another hook.
- the hook is called before the custom fields are added to $context['columns']

I'd say cut down everything and just one hook.
integrate_memberlist_buttons doesn't exist in 2.0, so there is no reason to keep it untouched, just rename it to something more generic. ;)

**Comment by norv** [comment]
Also a quick note: this PR is to 'development' branch. The 'development' branch has been left out-of-date.

If this is wanted, I would suggest to do something else first: either the author may want to PR to release-2.1 (or someone cherry-pick to it), either update the 'development' branch to the current head of release-2.1, then handle the PR.

**Comment by live627** [comment]
What is the point of this commit?

**Comment by Oldiesmann** [comment]
@live627 It's supposed to allow you to easily add extra columns to the memberlist from a mod or bridge.

**Comment by live627** [comment]
- See @emanuele45's comment.
- It can be done with integrate_memberlist_buttons

**Comment by Garthi** [comment]
hello, sorry i was a little absent in recent times, ok this pull was integrate a hook to integrate a new row with a mod without copy and paste manipulation, its a easy way to integrate a extra row, this code is not the best way however better than copy and paste installation method

what should i do next, a cherry-pick?
or this patch integrate to the release-2.1 brunch?

**Comment by live627** [comment]
@Arantor what are your thoughts on this?

**Comment by Arantor** [comment]
Firstly, yes, any patch should be to the release-2.1 branch. All evidence seems to be that we've essentially settled on using that one branch for everything for 2.1 at this point in time, which seems most productive right now as we're ultimately in a run-towards-beta mentality.

Secondly, are there likely to be many fields doing this and not using a custom profile field? (This, to me, would be a much more useful thing to add.)

Thirdly, if there's already a hook that exposes the relevant variables, I'd say use it at this stage. This will need to change in the future because we really shouldn't be exposing so much stuff through global scope and then hooks will be able to push items to hooked functions (so we'll need more hooks, but the benefit is better stability)

**Comment by live627** [comment]
Can this be now closed, @Oldiesmann?

**Comment by Oldiesmann** [comment]
Sounds good to me.

**Pull request closed without merging**